### PR TITLE
PBI 83: Register Account Page Integration

### PR DESCRIPTION
**Pull request by edutjie**
## Describe your changes
- Create Account schema
- Implement RTK Query for Register Account Page
- Add toast to display status after clicking register button
- Create UT for the integration
image
image

## Issue ticket number and link
- ADU-83: https://linear.app/adudu/issue/ADU-83/backend-integration

### CODE REVIEW

**Comment by edutjie** [comment]
Idk why, I think the Toast doesn't follow the styling of atom you implemented @FreeJ1nG, please help check this issue. 🙏

**Comment by FreeJ1nG** [comment]
> Idk why, I think the Toast doesn't follow the styling of atom you implemented @FreeJ1nG, please help check this issue. 🙏

Could you elaborate on what you mean here?

File: src/components/pages/register.tsx [hunk 1]
```diff
@@ -1,21 +1,79 @@
...
+    }
+    catch (error) {
+      toast({
+        variant: 'destructive',
+        title: 'Error',
+        description: (error as RegisterError).data.error,
+      })
+    }
+  }
+
```
Comment (FreeJ1nG) [inline src/components/pages/register.tsx]: Would you please merge from main and change this to just
```ts
  import { useErrorToast } from '@/lib/redux/hooks.ts'

  const [registerAccount, { isLoading, error }] = useRegisterAccountMutation()

  useErrorToast(error)

  const handleRegisterAccount = async (e: React.FormEvent<HTMLFormElement>) => {
    e.preventDefault()
    try {
      const { message } = await registerAccount({ payload: account }).unwrap()
      toast({
        title: 'Success',
        description: message,
      })
      setAccount(initialValue)
    }
    catch {
      // Already handled above
    }
  }
  ```

**Review by FreeJ1nG — CHANGES_REQUESTED** [review]

File: src/components/pages/register.tsx [hunk 2]
```diff
@@ -1,21 +1,79 @@
...
 import { Typography } from '@/components/atoms/typography.tsx'
 import { Footer } from '@/components/molecules/footer.tsx'
+import { useToast } from '@/hooks/use-toast.ts'
+import { useRegisterAccountMutation } from '@/lib/redux/features/account/api.ts'
+import { RegisterAccountRequestArgs, RegisterAccountResponse } from '@/lib/redux/features/account/validation.ts'
+
+interface RegisterError {
+  status: number
+  data: Partial<RegisterAccountResponse>
+}
```
Comment (ferryhieronimus) [inline src/components/pages/register.tsx]: This could be replaced with `FetchBaseQueryError`

Comment (ferryhieronimus) [reply src/components/pages/register.tsx]: However with Andrew's latest implementation this no longer needed

Comment (edutjie) [reply src/components/pages/register.tsx]: Okay thanks fer

**Comment by edutjie** [comment]
> > Idk why, I think the Toast doesn't follow the styling of atom you implemented @FreeJ1nG, please help check this issue. 🙏
> 
> Could you elaborate on what you mean here?

Never mind, yesterday I saw you implemented white bg for the toast, but it came out black in mine. I just realized that you also implemented dark mode.

**Comment by FreeJ1nG** [comment]
LGTM, will merge later

**Pull request merged**